src/components/BareMetalHosts/StatusComponents.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Popover } from '@patternfly/react-core';
+import { Icon } from 'patternfly-react';
+
+import { getHostStatus } from '../../utils/status/host/hostStatus';
+
+export const IconAndText = ({ icon, text }) => (
+  <React.Fragment>
+    <Icon type="pf" name={icon} className="co-icon-and-text__icon" />
+    {text}
+  </React.Fragment>
+);
+
+IconAndText.propTypes = {
+  icon: PropTypes.string.isRequired,
+  text: PropTypes.string.isRequired,
+};
+
+// Generic success status component
+export const GenericSuccess = ({ host }) => <IconAndText icon="ok" text={getHostStatus(host).text} />;
+
+GenericSuccess.propTypes = {
+  host: PropTypes.object.isRequired,
+};
+
+// Generic error status component
+// If the errorMessage property isn't empty its contents are
+// shown in a Popover.
+export const GenericError = ({ host }) =>
```

Comment:
To avoid multiple calls of the selector, the StatusComponents can take just host status object as prop which you can spread as `({ status, text, errorMessage }) => ...`

Reply:
Done